### PR TITLE
addedd enum for whereBirthRegistered, tests, updated request parameters, tests.

### PR DESCRIPTION
**Pull request by manishsw**
- Refactoring BRMS payload to use ENUM for WhereBirthRegistered

### CODE REVIEW

**Comment by manishsw** [comment]
Updated request parameters, refactor code with new loopup service to connect groconnector. Updated tests.

File: app/uk/gov/hmrc/brm/controllers/BirthEventsController.scala [hunk 1]
```diff
@@ -16,73 +16,73 @@
...
-                    """.stripMargin)
-
-                  respond(Ok(result))
-              }
-          )
-        } recover handleException("getReference")
-      )
-  }
+
+
```
Comment (adamconder) [inline app/uk/gov/hmrc/brm/controllers/BirthEventsController.scala]: Removing spacing

File: app/uk/gov/hmrc/brm/controllers/BirthEventsController.scala [hunk 2]
```diff
@@ -16,73 +16,73 @@
...
+         implicit request =>
+           request.body.validate[Payload].fold(
+             error => {
+               Logger.debug(s"[BirthEventsController][Connector][getReference] error: $error")
```
Comment (adamconder) [inline app/uk/gov/hmrc/brm/controllers/BirthEventsController.scala]: @manishsw Change this to Logger.info

File: app/uk/gov/hmrc/brm/services/LookupService.scala [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+}
+
+trait LookupService {
+
+  protected val groConnector : BirthConnector
+
+
+  private def getConnector(payload: Payload) : BirthConnector = {
+    payload.whereBirthRegistered match {
+      case BirthRegisterCountry.ENGLAND  =>
```
Comment (adamconder) [inline app/uk/gov/hmrc/brm/services/LookupService.scala]: Condense this into one case match

File: app/uk/gov/hmrc/brm/services/LookupService.scala [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+              }
+            }//end of getconnector
+          }
+        }
+
+
+      }
+
+
+
```
Comment (adamconder) [inline app/uk/gov/hmrc/brm/services/LookupService.scala]: @manishsw  Remove spacing

File: app/uk/gov/hmrc/brm/utils/BirthRegisterCountry.scala [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+object BirthRegisterCountry extends Enumeration  {
+
+  type BirthRegisterCountry = Value
+  val ENGLAND = Value("england")
+  val WALES = Value("wales")
+
+
+  def birthRegisterReads:Reads[BirthRegisterCountry]  = new Reads[BirthRegisterCountry] {
+    override def reads(json: JsValue): JsResult[BirthRegisterCountry.Value] = {
+      var countryValue = json.asOpt[String].get
```
Comment (adamconder) [inline app/uk/gov/hmrc/brm/utils/BirthRegisterCountry.scala]: Move this within the try block.
Could this be json.as[String]

File: conf/application.conf [hunk 6]
```diff
@@ -151,8 +151,8 @@ microservice {
         birth-registration-matching {
             host = localhost
             port = 9006
-            key = key
-            username = username
+            key = EUnm6wrsty/NmUhE3sTY
```
Comment (adamconder) [inline conf/application.conf]: Take this out.
Re-write your commit history into a single commit to remove this.

File: test/uk/gov/hmrc/brm/connectors/BirthConnectorSpec.scala [hunk 7]
```diff
@@ -49,10 +48,16 @@ class BirthConnectorSpec extends UnitSpec with WithFakeApplication with MockitoS
 
   object MockBirthConnector extends BirthConnector {
     override val httpGet = mockHttpGet
+    override val serviceUrl = ""
+    override val baseUri = ""
+    override val detailsUri = ""
   }
 
   object MockBirthConnectorInvalidUsername extends BirthConnector {
```
Comment (adamconder) [inline test/uk/gov/hmrc/brm/connectors/BirthConnectorSpec.scala]: Can you remove this object.

File: test/uk/gov/hmrc/brm/controllers/BirthEventsControllerSpec.scala [hunk 8]
```diff
@@ -67,261 +67,321 @@ class BirthEventsControllerSpec extends UnitSpec with WithFakeApplication with M
...
   def postRequest(v: JsValue) : FakeRequest[JsValue] = FakeRequest("POST", "/api/v0/events/birth")
     .withHeaders(("Content-type", "application/json"))
     .withBody(v)
 
+
+  object MockLookupService extends LookupService {
+    override val groConnector = mockConnector
+  }
+
   val mockConnector = mock[BirthConnector]
```
Comment (adamconder) [inline test/uk/gov/hmrc/brm/controllers/BirthEventsControllerSpec.scala]: Move this up above the mocklookupservice object

File: test/uk/gov/hmrc/brm/controllers/BirthEventsControllerSpec.scala [hunk 9]
```diff
@@ -67,261 +67,321 @@ class BirthEventsControllerSpec extends UnitSpec with WithFakeApplication with M
...
+      when(mockConnector.getReference(Matchers.any())(Matchers.any())).thenReturn(Future.successful(groJsonResponseObject))
+      val request = postRequest(userNoMatchIncludingReferenceNumber)
       val result = MockController.post().apply(request)
       (contentAsJson(result) \ "validated").as[Boolean] shouldBe false
       contentType(result).get shouldBe "application/json"
     }
 
-    "return JSON response of true on successful detail match" in {
-      when(MockController.Connector.getReference(Matchers.any())(Matchers.any())).thenReturn(Future.successful(groJsonResponseObject))
+   /* "return JSON response of true on successful detail match" in {
```
Comment (adamconder) [inline test/uk/gov/hmrc/brm/controllers/BirthEventsControllerSpec.scala]: @manishsw Is this unit test needed?

File: app/uk/gov/hmrc/brm/controllers/BirthEventsController.scala (hunk 1 above)
Comment (manishsw) [reply app/uk/gov/hmrc/brm/controllers/BirthEventsController.scala]: done

File: app/uk/gov/hmrc/brm/controllers/BirthEventsController.scala (hunk 2 above)
Comment (manishsw) [reply app/uk/gov/hmrc/brm/controllers/BirthEventsController.scala]: done

File: app/uk/gov/hmrc/brm/services/LookupService.scala (hunk 3 above)
Comment (manishsw) [reply app/uk/gov/hmrc/brm/services/LookupService.scala]: done

File: app/uk/gov/hmrc/brm/services/LookupService.scala (hunk 4 above)
Comment (manishsw) [reply app/uk/gov/hmrc/brm/services/LookupService.scala]: done

File: app/uk/gov/hmrc/brm/utils/BirthRegisterCountry.scala (hunk 5 above)
Comment (manishsw) [reply app/uk/gov/hmrc/brm/utils/BirthRegisterCountry.scala]: updated it.

File: conf/application.conf (hunk 6 above)
Comment (manishsw) [reply conf/application.conf]: done

File: test/uk/gov/hmrc/brm/connectors/BirthConnectorSpec.scala (hunk 7 above)
Comment (manishsw) [reply test/uk/gov/hmrc/brm/connectors/BirthConnectorSpec.scala]: done

File: test/uk/gov/hmrc/brm/controllers/BirthEventsControllerSpec.scala (hunk 8 above)
Comment (manishsw) [reply test/uk/gov/hmrc/brm/controllers/BirthEventsControllerSpec.scala]: done

File: test/uk/gov/hmrc/brm/controllers/BirthEventsControllerSpec.scala (hunk 9 above)
Comment (manishsw) [reply test/uk/gov/hmrc/brm/controllers/BirthEventsControllerSpec.scala]: removed it.

**Pull request closed without merging**